### PR TITLE
add some new beans and github api methods to retrieve Project, Columns, Cards

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

### CODE REVIEW

**Comment by joakime** [comment]
I like it.
I pushed a few changes to the ListIterator stuff tho.

File: pom.xml [hunk 1]
```diff
@@ -36,9 +36,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>ch.qos.logback</groupId>
-      <artifactId>logback-classic</artifactId>
-      <version>1.2.3</version>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.30</version>
```
Comment (joakime) [inline pom.xml:41]: This change already exists in main.

File: pom.xml [hunk 2]
```diff
@@ -56,6 +56,12 @@
       <version>5.5</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>1.7.30</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (joakime) [inline pom.xml:64]: This change already exists in main (but using logback instead, so we can occasionally turn on DEBUG level, which slf4j-simple doesn't support).

File: src/main/java/net/webtide/tools/github/Card.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public ZonedDateTime getUpdatedAt()
+    {
+        return updatedAt;
+    }
+
+    public Issue getIssue()
+    {
+        try
+        {
+            return GitHubApi.connect().query(content_url, Issue.class, builder -> builder.GET()
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/Card.java:47]: I'm not a fan of this here.
Are we doing this in other beans?

File: src/main/java/net/webtide/tools/github/GitHubApi.java [hunk 4]
```diff
@@ -289,19 +295,42 @@ public PullRequestCommits pullRequestCommits(String repoOwner, String repoName,
...
     }
 
-    public Releases listReleases(String repoOwner, String repoName, int resultsPerPage, int pageNum) throws IOException, InterruptedException
+    public Projects listProjects(String repoOwner, String repoName, int resultsPerPage, int pageNum) throws IOException, InterruptedException
     {
         Query query = new Query();
-        query.put("per_page", String.valueOf(resultsPerPage));
-        query.put("page", String.valueOf(pageNum));
+        if (resultsPerPage > 0) query.put("per_page", String.valueOf(resultsPerPage));
+        if (pageNum > 0) query.put("page", String.valueOf(pageNum));
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/GitHubApi.java:302]: Are these really optional ??

File: src/main/java/net/webtide/tools/github/Projects.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+// ========================================================================
+//
+
+package net.webtide.tools.github;
+
+import java.util.ArrayList;
+
+public class Projects extends ArrayList<Project>
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/Projects.java:17]: You'll want a ListIterator for this as well.

File: src/main/java/net/webtide/tools/github/User.java [hunk 6]
```diff
@@ -83,4 +83,13 @@ public boolean isSiteAdmin()
     {
         return permissions;
     }
+
+    @Override
+    public String toString()
+    {
+        return "User{" + "login='" + login + '\'' + ", type='" + type + '\'' + ", name='" + name + '\'' + ", company='"
+            + company + '\'' + ", location='" + location + '\'' + ", email='" + email + '\'' + ", bio='" + bio + '\''
+            + ", siteAdmin=" + siteAdmin + ", permissions=" + permissions + ", createdAt=" + createdAt + ", updatedAt="
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/User.java:92]: I would rather see this toString be smarter, and not include attributes that are null/unset in it's output

File: src/main/java/net/webtide/tools/github/Columns.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+// ========================================================================
+//
+
+package net.webtide.tools.github;
+
+import java.util.ArrayList;
+
+public class Columns extends ArrayList<Column>
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/Columns.java:17]: ListIterator for this too.

File: src/main/java/net/webtide/tools/github/Cards.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+// ========================================================================
+//
+
+package net.webtide.tools.github;
+
+import java.util.ArrayList;
+
+public class Cards extends ArrayList<Card>
```
Comment (joakime) [inline src/main/java/net/webtide/tools/github/Cards.java:17]: ListIterator for this too.

**Review by joakime — CHANGES_REQUESTED** [review]
Some notes

File: src/main/java/net/webtide/tools/github/Card.java (hunk 3 above)
Comment (olamy) [reply src/main/java/net/webtide/tools/github/Card.java:51]: removed

**Pull request merged**